tests/unit/test_shape_decoders.py

```diff
@@ -277,3 +281,53 @@ def test_decoder(self, opt_unit_label):
         assert v.x.val == 1.0
         assert v.y.__class__ is RDoubleI
         assert v.y.val == 2.0
+
+
+TRANSFORMATIONS = [
+    (
+        'matrix(1.0 0.0 0.0 1.0 0.0 0.0)',
+        'matrix(1.0 0.0 0.0 1.0 0.0 0.0)',
+    ),
+    (
+        'none',
+        'none',
+    ),
+    (
+        'translate(3 4)',
+        'matrix(1.0 0.0 0.0 1.0 3.0 4.0)',
+    ),
+    (
+        'translate(5)',
+        'matrix(1.0 0.0 0.0 1.0 5.0 0.0)',
+    ),
+    (
+        'scale(1.5 2.5)',
+        'matrix(1.5 0.0 0.0 2.5 0.0 0.0)',
+    ),
+    (
+        'scale(1.5)',
+        'matrix(1.5 0.0 0.0 1.5 0.0 0.0)',
+    ),
+    (
+        'rotate(0.785398163397448)',
```

Comment:
Maybe pick an angle of rotation where `cos(a) != sin(a)` to have this be as thorough as possible?